File: ProyectovisualBasic/ProyectovisualBasic/Pilas.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace Proyecto_Final1

    Public Class Pilas
        Inherits Form

        Private pila As Pila

        Public Sub New()
            MyBase.New
            InitializeComponent
            Me.pila = New Pila
        End Sub

        Private Sub btnPush_Click(ByVal sender As Object, ByVal e As EventArgs)
            Dim n As Nodo
            Dim d As Integer = Integer.Parse(textBox1.Text)
            n = New Nodo
            n.Dato = d
            n.Siguiente = Nothing
            Me.pila.Push(n)
            txtDato.Clear
            txtDato.Focus
            listBox1.Items.Add(d)
        End Sub

        Private Sub btnPop_Click(ByVal sender As Object, ByVal e As EventArgs)
            If Me.pila.pila_vacia Then
                MessageBox.Show("pila vacia")
                Return
            End If

            Dim n As Nodo
            n = Me.pila.Pop
            MessageBox.Show(("salio " + n.Dato))
            Dim a As Integer
            a = n.Dato
            listBox1.Items.Remove(a)
        End Sub

        Private Sub label1_Click(ByVal sender As Object, ByVal e As EventArgs)

        End Sub

        Private Sub listBox1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs)

        End Sub

        Private Sub txtDato_TextChanged(ByVal sender As Object, ByVal e As EventArgs)

        End Sub

        Private Sub label2_Click(ByVal sender As Object, ByVal e As EventArgs)

        End Sub

        Private Sub InitializeComponent()
            Me.SuspendLayout()
            '
            'Pilas
            '
            Me.ClientSize = New System.Drawing.Size(278, 244)
            Me.Name = "Pilas"
            Me.ResumeLayout(False)

        End Sub
    End Class
End Namespace

File: Proyecto Riojas/Proyecto Final1/Proyecto Final1/Pilas.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace Proyecto_Final1
{
    public partial class Pilas : Form
    {
        Pila pila;
        public Pilas()
        {
            InitializeComponent();
            pila = new Pila();
        }

        private void btnPush_Click(object sender, EventArgs e)
        {
            Nodo n;

            int d = int.Parse(txtDato.Text);
            n = new Nodo();
            n.Dato = d;
            n.Siguiente = null;

            pila.Push(n);
            txtDato.Clear();
            txtDato.Focus();
            listBox1.Items.Add(d);

        }

        private void btnPop_Click(object sender, EventArgs e)
        {
            if (pila.pila_vacia())
            {
                MessageBox.Show("pila vacia");
                return;

            }
            Nodo n;
            n = pila.Pop();

            MessageBox.Show("salio " + n.Dato);
            int a;
            a = n.Dato;
            listBox1.Items.Remove(a);
        }

    private void label1_Click(object sender, EventArgs e)
    {

    }

    private void listBox1_SelectedIndexChanged(object sender, EventArgs e)
    {

    }

    private void txtDato_TextChanged(object sender, EventArgs e)
    {

    }

    private void label2_Click(object sender, EventArgs e)
    {

    }
  }
}
